### DISCUSSION TITLE
build error: micromdm/scep checksum mismatch when building with cgo / PKCS11

### DISCUSSION THREAD

**Discussion by jasonhernandez** [smallstep/certificates#635] [Q&A] [answered]
I'm attempting to build certificates with PKCS11 support using cgo, following https://github.com/smallstep/certificates/blob/master/docs/CONTRIBUTING.md#build-step-ca-using-cgo

I get an error following this command (on multiple attempts): `make bootstrap && make build GOFLAGS=""`

go: verifying github.com/micromdm/scep/v2@v2.0.0: checksum mismatch
	downloaded: h1:imFhkuhyLFXXVwiqNerGq7hbrLjGatoX+/YzGTA85Ik=
	go.sum:     h1:cRzcY0S5QX+0+J+7YC4P2uZSnfMup8S8zJu/bLFgOkA=

Every other package seems to pass verification. Is there something wrong with this dependency? Is there anything else I might be missing or that might be helpful for troubleshooting?
I've cleared out the repo and ~/go and tried this a few times with the same error.

This is on a fresh Debian Stable (10 / buster) amd64 VM with build-essential, gcc, go, and libpcsclite-dev installed.

Thanks for any assistance

**Comment by maraino** [comment, marked as answer]
Hi @jasonhernandez, what version of Go and what version of certificates are you trying to build? 

I've started a VM with Debian 10 (Buster) using "go version go1.16.3 linux/amd64" and the latest master branch (b9743b36e1ba2b17687181d036467cb604469a84) on certificates and it builds with the instructions that you pointed, and all modules are verified if I run "go mod verify".

**Reply by jasonhernandez** [reply]
Thanks for the help - it looks like the version of Go was the problem.
I ran into this when I pull the latest version of master on commit b9743b3, with the default Go compiler in Debian Buster.
> go version go1.11.6 linux/amd64

When I use the Go compiler from Buster-backports - go1.15.9 linux/amd64, the build succeeds.

**Reply by maraino** [reply]
We only support 1.15+
